nuxeo-services/nuxeo-platform-usermanager-api/src/main/java/org/nuxeo/ecm/platform/usermanager/UserManager.java

```diff
@@ -58,6 +58,12 @@
      */
     NuxeoPrincipal getPrincipal(String username);
 
+    /**
+     * Calls {@link #getPrincipal(String)} and controls if the principal
+     * references (groups, etc) have to be fetched.
+     */
+    NuxeoPrincipal getPrincipal(String username, boolean fetchReferences);
```

Comment:
Maybe add in the Javadoc of `NuxeoPrincipal getPrincipal(String username)` that it fetches the references?